api/db/changelog/db.changelog-44-user-recent-resource.xml

```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog/1.9"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog/1.9
+                    http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-1.9.xsd">
+    <changeSet author="nehasaxena" id="db.changelog-44-cacheTable">
+        <createTable tableName="user_recent_resource">
+            <column name="id" type="bigint" autoIncrement="true">
```

Comment:
I'm not a MySQL expert, but in general having a globally increasing ID synchronized across all users (which would otherwise be independent) doesn't seem like it would have ideal performance.

Reading a few threads about this online indicates that this shouldn't be a major concern so I'm probably fine going with what you have here, but I'm curious if others have experience/thoughts on this. The alternative would be to use a GUID or just a random int64

Reply:
In practice, given that this table will be limited in size per user, any choice of ID datatype should be OK. In other systems with many hundreds of thousands of rows, we've seen real problems with GUIDs - indexing became a bottleneck. Random ints and auto incremented ids shouldn't be a problem.

Reply:
To clarify, I'm not concerned with size or indexing overhead. Mainly my worry was around having a synchronized step when inserting two independent rows (which is highly likely to happen concurrently given the use case). I can imagine such synchronization might be a property of any MySQL row insertion anyways though, so making it autoincrement might have very little overhead.

Reply:
👍